limitador/src/storage/redis/redis_cached.rs

```diff
@@ -240,15 +244,30 @@ impl CachedRedisStorage {
         let batcher: Arc<Mutex<HashMap<Counter, AtomicExpiringValue>>> =
             Arc::new(Mutex::new(Default::default()));
 
+        let (flush_tx, mut flush_rx) = tokio::sync::watch::channel(());
+
         {
             let storage = async_redis_storage.clone();
             let counters_cache_clone = counters_cache.clone();
             let conn = redis_conn_manager.clone();
             let p = Arc::clone(&partitioned);
             let batcher_flusher = batcher.clone();
-            let mut interval = tokio::time::interval(flushing_period);
+
             tokio::spawn(async move {
                 loop {
+                    // Wait for a new flush request,
+                    flush_rx.changed().await.unwrap();
+
+                    if flushing_period != Duration::ZERO {
+                        // Set the flushing_period to reduce the load on Redis the downside to
+                        // setting it the cached counters will not be as accurate as when it's zero.
+                        tokio::time::sleep(flushing_period).await
+                    }
```

Comment:
Have you looked at tokio's [`select!`](https://docs.rs/tokio/latest/tokio/macro.select.html)? We could have either wait for the period to elapse _or_ wait to be notified... So that you'd flush _every n seconds_ or flush when notified. Cause this would have the flush task wait _after_ being notified.

Reply:
See [here](https://github.com/Kuadrant/limitador/pull/294#discussion_r1579466230)... I don't know how much of an added value that notification is useful really... What do you think we're addressing by introducing this behavior? Other than the loop as fast as possible with the `0` flushing period... This "just" changes what the starting point of a flushing period is... 😕 What am I missing?

Reply:
The [tokio::select!](https://docs.rs/tokio/latest/tokio/macro.select.html#) would be useful if we want to _flush_ when whatever happens first, either the notify() or the flushing_period.